core/network/corda-interop-app/interop-workflows/src/main/kotlin/com/weaver/corda/app/interop/flows/AssetTransferFlows.kt

```diff
@@ -0,0 +1,859 @@
+ * @property claimStatusLinearId The unique identifier of the vault state representing asset claim details fetched via interop-query.
+ * @property issuer The issuing authority of the pledged fungible asset, if applicable (e.g., fungible house token).
+ *                      Otherwise this will be same as the party submitting the transaction.
+ * @property observers The parties who are not transaction participants but only observers (can be empty list).
+ */
+object ReclaimPledgedAsset {
+    @InitiatingFlow
+    @StartableByRPC
+    class Initiator
+    @JvmOverloads
+    constructor(
+        val pledgeId: String,
+        val assetStateCreateCommand: CommandData,
+        val claimStatusLinearId: String,
+        val issuer: Party,
+        val observers: List<Party> = listOf<Party>()
+    ) : FlowLogic<Either<Error, SignedTransaction>>() {
+        /**
+         * The call() method captures the logic to create a new asset state (that was redeemed earlier via a 
+         * pledge transaction) by consumes the [AssetPledgeState].
+         *
+         * @return Returns SignedTransaction.
+         */
+        @Suspendable
+        override fun call(): Either<Error, SignedTransaction> = try {
+            val linearId = getLinearIdFromString(pledgeId)
+
+            val viewData = subFlow(GetExternalStateByLinearId(claimStatusLinearId))
+            val externalStateView = ViewDataOuterClass.ViewData.parseFrom(viewData)
+            val payloadDecoded = Base64.getDecoder().decode(externalStateView.payload.toByteArray())
+            val assetClaimStatus = AssetTransfer.AssetClaimStatus.parseFrom(payloadDecoded)
+            println("Asset claim status details obtained via interop query: ${assetClaimStatus}")
+
+            val assetPledgeStateAndRef = subFlow(GetAssetPledgeStateById(pledgeId))
+            if (assetPledgeStateAndRef == null) {
+                println("AssetPledgeState for Id: ${linearId} not found.")
+                Left(Error("AssetPledgeState for Id: ${linearId} not found."))
+            } else {
+                val assetPledgeState = assetPledgeStateAndRef.state.data
+                println("Party: ${ourIdentity} ReclaimPledgeState: ${assetPledgeState}")
+
+                if (assetClaimStatus.claimStatus) {
+                    println("Cannot perform reclaim for pledgeId $pledgeId as the asset was claimed in remote network.")
+                    Left(Error("Cannot perform reclaim for pledged $pledgeId as the asset was claimed in remote network."))
+                } else if (!assetClaimStatus.expirationStatus) {
+                    println("Cannot perform reclaim for pledgeId $pledgeId as the asset pledge has not yet expired.")
+                    Left(Error("Cannot perform reclaim for pledged $pledgeId as the asset pledge has not yet expired."))
+                } else if (assetClaimStatus.expiryTimeSecs != assetPledgeState.expiryTimeSecs) {
```

Comment:
Can this condition be first the one among three above, because if the expiryTimeSecs doesn't match but according to claimStatus expirationStatus is true, this condition isn't checked at all.

Reply:
There are three conditions in the if-else blocks. Only if all the 3 are not correct, we go to the last else block. So I didn't get the scenario under which re-ordering these conditions improves the situation.

Reply:
Take an example. assetClaimStatus.expiryTimeSecs = assetPledgeState.expiryTimeSecs + 100, hence assetClaimStatus.expirationStatus=false, but as per assetPledgeState.expiryTimeSecs, it is already expired 100 secs early.
Now going by this sequence of conditions, we get message : `pledge has not expired yet`. 
But when we get `AssetPledgeState`, we see that pledge is supposed to be expired, so people can get confused why system is behaving wrong here because of the message we see is wrong. And as corda-interop-app is core component so we can't ignore this kind of wrong message being delivered to the user.

This is the reason, why I was suggesting to re-order, so that first we check the correctness of values, and if that is wrong, we flag that values don't match, after that we decide based upon that value whether pledge has expired or not.
I hope this clarifies.

Reply:
@sanvenDev Ideally, the value of the `expiryTimeSecs` should be the same on both the networks. However, as you are pointing out, there may be some difference in the exporting and importing networks due to say clock drift. In this case, either the claimer in the importing network or pledger in the exporting network should get penalized. I.e., either pledger need to wait more time before re-claim or claimer need to claim well before the expiry time itself. With the above implementation, we see that the pledger gets penalized (i.e., need to wait more time before the pledger can re-claim). Hence, writing the three conditions in any order doesn't make any difference. If you re-order the conditions, it results into penalizing the claimer.

Reply:
No, the order won't change any effect on outcome. It will only affect what `log message` you are providing to the user. Please go through my previous message again to understand why the log message will be wrong.

And if the value was guaranteed to be same on both networks, then why do we even need the 3rd condition?

I still feel providing wrong `log message` from core apps should be avoided. If you think wrong `log messages` doesn't matter sure we can ignore this.

Reply:
Done.